src/Metrics/CarbonSink.cpp

```diff
@@ -0,0 +1,15 @@
+#include "CarbonSink.h"
+#include "InternalMetric.h"
+
+namespace Metrics {
+
+void CarbonSink::reportMetric(InternalMetric &MetricToBeReported) {
+  auto CurrentValue = MetricToBeReported.Counter->load();
```

Comment:
We are mixing memory ordering here (we are using relaxed memory ordering elsewhere). I am assuming that this has not effect on atomicity (is that a word?), do you know?

Reply:
My guess is that this would force sequential everywhere. No reason not to continue with relaxed, so I'll specify it here (and in the log sink) and add a note to the Sink interface for people adding new sinks in the future.